File: packages/prebid_mobile_sdk_max/android/src/main/kotlin/com/prebid/prebid_mobile_sdk_max/PrebidMobileSdkMaxPlugin.kt
package com.prebid.prebid_mobile_sdk_max

import android.app.Activity
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.embedding.engine.plugins.activity.ActivityAware
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding

/// Companion plugin that adds AppLovin MAX mediation on top of the core
/// prebid_mobile_sdk plugin. Registers the MAX banner PlatformView factory and
/// the MAX interstitial method channel.
class PrebidMobileSdkMaxPlugin : FlutterPlugin, ActivityAware {

    private var activity: Activity? = null
    private var interstitialManager: MaxInterstitialManager? = null
    private var rewardedManager: MaxRewardedManager? = null

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        binding.platformViewRegistry.registerViewFactory(
            "prebid_mobile_sdk_max/banner",
            MaxBannerAdViewFactory(binding.binaryMessenger) { activity },
        )
        binding.platformViewRegistry.registerViewFactory(
            "prebid_mobile_sdk_max/native",
            MaxNativeAdViewFactory(binding.binaryMessenger) { activity },
        )
        interstitialManager = MaxInterstitialManager(binding.binaryMessenger) { activity }
        rewardedManager = MaxRewardedManager(binding.binaryMessenger) { activity }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        interstitialManager?.dispose()
        interstitialManager = null
        rewardedManager?.dispose()
        rewardedManager = null
    }

    override fun onAttachedToActivity(binding: ActivityPluginBinding) {
        activity = binding.activity
    }

    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
        activity = binding.activity
    }

    override fun onDetachedFromActivityForConfigChanges() {
        activity = null
    }

    override fun onDetachedFromActivity() {
        activity = null
    }
}

File: packages/prebid_mobile_sdk_max/ios/prebid_mobile_sdk_max/Sources/prebid_mobile_sdk_max/PrebidMobileSdkMaxPlugin.swift
import Flutter
import UIKit

/// Companion plugin that adds AppLovin MAX mediation on top of the core
/// prebid_mobile_sdk plugin. Registers the MAX banner PlatformView factory and
/// the MAX interstitial method channel.
public class PrebidMobileSdkMaxPlugin: NSObject, FlutterPlugin {

    /// Retains the interstitial and rewarded managers for the plugin's lifetime.
    private static var interstitialManager: MaxInterstitialManager?
    private static var rewardedManager: MaxRewardedManager?

    public static func register(with registrar: FlutterPluginRegistrar) {
        let bannerFactory = MaxBannerAdViewFactory(messenger: registrar.messenger())
        registrar.register(bannerFactory, withId: "prebid_mobile_sdk_max/banner")

        let nativeFactory = MaxNativeAdViewFactory(messenger: registrar.messenger())
        registrar.register(nativeFactory, withId: "prebid_mobile_sdk_max/native")

        interstitialManager = MaxInterstitialManager(messenger: registrar.messenger())
        rewardedManager = MaxRewardedManager(messenger: registrar.messenger())
    }
}
